app/uk/gov/hmrc/fileupload/testonly/TestOnlyController.scala

```diff
@@ -82,8 +83,13 @@ class TestOnlyController(baseUrl: String, wSClient: WSClient, val s3Service: S3J
     }
   }
 
-  def transferGetEnvelopes() = Action.async { implicit request =>
-    wSClient.url(s"$baseUrl/file-transfer/envelopes").get().map { response =>
+  def transferGetEnvelopes(destination: Option[String]) = Action.async { implicit request =>
+    val transferUrl = s"$baseUrl/file-transfer/envelopes"
+    val wsUrl = destination match {
+      case Some(d)  => wSClient.url(transferUrl).withQueryString(("destination", URLEncoder.encode(d, "UTF-8")))
```

Comment:
You can simplify this a bit if you want as we always want `wSClient.url(transferUrl)` (so the fold would be over just the withQueryString)